### ISSUE TITLE
Leaderboard App crash at SplashScreen

### ISSUE THREAD

**Issue by Jahz64** [playgameservices/play-games-plugin-for-unity#2238]
Hello,

I'm having some issues setting up leaderboard in Unity. I buy an app who needed to be reskin everything work perfectly expect leaderboard here is the script i use for Google play leaderboard

`
using UnityEngine;
using System.Collections;
using UnityEngine.SceneManagement;

#if GooglePlayDef
using GooglePlayGames;        
#endif

public class GooglePlayManager : MonoBehaviour
{

	public static GooglePlayManager singleton;

	private AudioSource sound;

	[HideInInspector]
	public managerVars vars;

	void OnEnable()
	{
		vars = Resources.Load<managerVars>("managerVarsContainer");
		SceneManager.sceneLoaded += OnLevelFinishedLoading;
	}

	void Awake()
	{
		MakeInstance();
	}

	void MakeInstance()
	{
		if (singleton != null)
		{
			Destroy(gameObject);
		}
		else
		{
			singleton = this;
			DontDestroyOnLoad(gameObject);
		}
	}

	// Use this for initialization
	void Start()
	{
		sound = GetComponent<AudioSource>();
		#if GooglePlayDef
		PlayGamesPlatform.Activate();
		Social.localUser.Authenticate((bool success) =>
			{
				if (success)
				{

				}
			});
		#endif
	}

	void OnLevelFinishedLoading(Scene scene, LoadSceneMode mode)
	{
		ReportScore(GameManager.instance.currentScore);
	}

	public void OpenLeaderboardsScore()
	{
		#if GooglePlayDef
		if (Social.localUser.authenticated)
		{
			PlayGamesPlatform.Instance.ShowLeaderboardUI(vars.leaderBoardID);
		}
		#endif
	}

	void ReportScore(int score)
	{
		if (Social.localUser.authenticated)
		{
			Social.ReportScore(score, vars.leaderBoardID, (bool success) => { });
		}
	}

}
`
Here is the problem 

	// Use this for initialization
	void Start()
	{
		sound = GetComponent<AudioSource>();
		#if GooglePlayDef
		PlayGamesPlatform.Activate();
		Social.localUser.Authenticate((bool success) =>
			{
				if (success)
				{

				}
			});
		#endif
	}

Those lines were commented but when I uncomment it my app just crash after the splashscreen

I'm using the last version of playgameservices 0.9.50 
The last version of SDK
I'm also using Google Mobile Ads Unity Plugin

Everything is updated with Android Studio

I have tried this solution but it doesn't seems to work 

https://answers.unity.com/questions/1324988/android-game-crashes-on-startup-becaus-of-google-p.html

Everything is setup correctly but those lines in "void Start" if uncommented make my game crash on splashscreen maybe i'm missing something in it ? or there is some sort of conflict but I don't know why and where it is 

I'm not a good programmer but I like to play around with the code, but here i'm really lost..

Google Play Dev :
http://prntscr.com/k4qooj
http://prntscr.com/k4qpqd

In Unity : 
http://prntscr.com/k4qsye

Tell me if you have any question. Thank you !

**Comment by shyamPindoria** [comment]
It seems there is an incompatibility between the GooglePlayGames plugin and version 10.2.0 of the Google Play Services. So the best thing to do is roll back Google Play Services to 10.0.1
#1604
The full steps are from tolgau on the above thread.

**Comment by Jahz64** [comment]
Thank you for taking the time to respond

by rolling back you mean just replace all ARR files ? because i use the 12.0.1 not the 10.2.0 ?

**Comment by shyamPindoria** [comment]
Yes, replacing files manually as suggested by tolgau in this thread #1604

I also use the 12.0.1. I haven't tried rolling back because I am assuming the problem is something else as I started getting problems just after updating to the latest Unity 2018.2.0f from 2018.1.7f where it was working properly with GPGS 12.0.1.

**Comment by hytka81** [comment]
My game also started to crash at startup when doing silent login after upgrading to latest Unity 2018.2.0f2. 

I'm currently using:

### Google Play Games Plugin
- GooglePlayGamesPlugin-0.9.50.unitypackage
### From Tools:
- Android SDK Tools Rev 25.2.5
- Android SDK Platform-tools Rev 28
- Android SDK Build-tools Rev 28.0.1
### From Extras:
- Android Support Repository Rev 47
- Android Support Library Rev 23.2.1
- Google Play services Rev 49

This is little bit annoying since it is now preventing releasing an update for my game :(

So do I need to roll back to some previous version of the Google Play Games Plugin? Little bit confused with these GPGS versions...

**Comment by shyamPindoria** [comment]
I’d suggest you to roll back your Unity version instead of rolling back GPGS.
Maybe try the recently released version 2018.1.8.

**Comment by Jahz64** [comment]
Thanks i'll try to rollback to 10.0.1 since I'm using an old version of Unity 5.5.3p4. I'am a bit afraid to update it and lose something or getting more errors.
I manage to get all things work expect leaderboard so i'll come back to you if rolling back to 10.0.1 fixed it have a good day !

**Comment by shyamPindoria** [comment]
I just rolled back from Unity 2018.2.0f2 to 2018.1.8f1 and its **now working** with Play Services 12.0.1.
Hopefully  @unity-technologies fixes this issue in upcoming versions.

**Comment by hytka81** [comment]
I was able to get this working using fix recommended in [#2013](https://github.com/playgameservices/play-games-plugin-for-unity/issues/2013)

**Comment by SaadAli2016** [comment]
**FINALLY fIXED IT** 
turns out the two plugins are not compatible here is the fix i made a video on that.
[https://youtu.be/L-N7Tsw6Jf4
![YoutubeThumbnail](https://user-images.githubusercontent.com/24651656/67547504-b2a5d900-f718-11e9-8ad2-04a014b3c54b.png)
](url)

**Comment by vaibhavgpt-google** [comment]
Closing this issue as part of a cleanup of older tickets. Google Play Games v1 is now deprecated. Please upgrade to version v2.1.0 or newer: https://github.com/playgameservices/play-games-plugin-for-unity/releases/tag/v2.1.0. If this issue persists on the new v2 architecture, please open a new ticket.
